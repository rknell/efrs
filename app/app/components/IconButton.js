import React, { Component } from 'react'
import { ActivityIndicator, Text, TouchableOpacity, View } from 'react-native'
import IconFa from 'react-native-vector-icons/FontAwesome'
import IconIonicon from 'react-native-vector-icons/Ionicons'
import GlobalStyle from '../services/GlobalStyle'
import Translate from '../services/Translate'

export default class IconButton extends Component {
  getIcon = () => {
    if (this.props.name) {
      if (this.props.iconset === 'ionicons') {
        return <IconIonicon name={this.props.name} size={18} color={GlobalStyle.style.primaryColourText} />
      } else {
        return <IconFa name={this.props.name} size={18} color={GlobalStyle.style.primaryColourText} />
      }
    } else {
      return null
    }
  }

  render () {
    return (
      <TouchableOpacity onPress={this.props.onPress} disabled={this.props.isDisabled}>
        <View style={[{
          padding: 6,
          minWidth: 30,
          alignItems: 'center',
          justifyContent: 'center',
          borderRadius: 2,
          backgroundColor: this.props.isDisabled ? GlobalStyle.style.primaryColourDisabled : GlobalStyle.style.primaryColour,
          flexDirection: 'row'
        }, this.props.style]}>
          {this.getIcon()}
          {!!this.props.label && !this.props.isBusy &&
          <Text style={{ fontSize: 18, color: GlobalStyle.style.primaryColourText }}>&nbsp;{this.props.translate ? Translate.do(this.props.label) : this.props.label}</Text>
          }
          {!!this.props.isBusy && <ActivityIndicator />}
        </View>
      </TouchableOpacity>
    )
  }
}
