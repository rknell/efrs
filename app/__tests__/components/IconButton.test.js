import React from 'react'
import IconButton from '../../app/components/IconButton'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

it('should render', () => {
  const wrapper = shallow(<IconButton />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should render a label', () => {
  const wrapper = shallow(<IconButton label='TEST LABEL' />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should show a ui change when disabled', () => {
  const wrapper = shallow(<IconButton isDisabled />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
