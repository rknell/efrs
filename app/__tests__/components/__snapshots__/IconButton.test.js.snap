// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "dodgerblue",
          "borderRadius": 2,
          "flexDirection": "row",
          "justifyContent": "center",
          "minWidth": 30,
          "padding": 6,
        },
        undefined,
      ]
    }
  />
</TouchableOpacity>
`;

exports[`should render a label 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "dodgerblue",
          "borderRadius": 2,
          "flexDirection": "row",
          "justifyContent": "center",
          "minWidth": 30,
          "padding": 6,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
          "fontSize": 18,
        }
      }
    >
      Â 
      TEST LABEL
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`should show a ui change when disabled 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  disabled={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(30, 144, 255, 0.3)",
          "borderRadius": 2,
          "flexDirection": "row",
          "justifyContent": "center",
          "minWidth": 30,
          "padding": 6,
        },
        undefined,
      ]
    }
  />
</TouchableOpacity>
`;
